pkg/guardian/blacklist_test.go

```diff
@@ -0,0 +1,94 @@
+package guardian
+
+import (
+	"context"
+	"net"
+	"testing"
+)
+
+type FakeBlacklistStore struct {
+	blacklist []net.IPNet
+}
+
+func (f FakeBlacklistStore) GetBlacklist() []net.IPNet {
+	return f.blacklist
+}
+
+func TestIsBlacklisted(t *testing.T) {
+	store := &FakeBlacklistStore{}
+	blacklister := NewIPBlacklister(store, TestingLogger, NullReporter{})
+
+	tests := []struct {
+		// test case setup
+		name           string
+		storeBlacklist []net.IPNet
+		req            Request
+
+		// expected results
+		blacklisted bool
+		errored     bool
+	}{
+		{
+			name:           "Blacklisted",
+			storeBlacklist: parseCIDRs([]string{"10.0.0.1/24"}),
```

Comment:
It's actually a range. 10.0.0.1/24 is still valid though just unconventional, right?

Reply:
This format is usually not a subnet definition but an IP address definition. You're using CIDR notation to specify the subnet mask instead of doing `10.0.0.1/255.255.255.0`. If it is a CIDR then I suggest using the actual network address of the subnet instead of this IP address just for convention. But really it isn't a big deal.